### ISSUE TITLE
GM_xmlhttpRequest form-data field order error

### ISSUE THREAD

**Issue by lisonge** [Tampermonkey/tampermonkey#1783]
## Expected Behavior
![image](https://github.com/Tampermonkey/tampermonkey/assets/38517192/edae8df0-f939-479e-88c2-584457bbc5a6)

`file` field is last one 

## Actual Behavior

![image](https://github.com/Tampermonkey/tampermonkey/assets/38517192/a9e94291-7a48-4b34-9c20-89e28f2a9c27)

`file` field is frist one 

## Specifications

- Chrome/113.0.0.0
- TM: 4.18.1
- OS: Win11

## Script

```js
// ==UserScript==
// @name MyName
// @namespace http://mywebsite.com
// @Version 1.0
// @description try to take over the world!
// @author Me
// @match http://mywebsite.com/*
// @include https://mywebsite.com/*
// @grant GM_xmlhttpRequest
// @connect *
// ==/UserScript==

const imgUrl = `data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAIAAACQd1PeAAAAAXNSR0IArs4c6QAAAARnQU1BAACxjwv8YQUAAAAJcEhZcwAAHYcAAB2HAY/l8WUAAAAMSURBVBhXY/j//z8ABf4C/qc1gYQAAAAASUVORK5CYII=`;
const blob = await fetch(imgUrl).then((r) => r.blob());
const data = new FormData();
data.append(`key`, `value`);
data.append('file', blob, `name.png`);
GM_xmlhttpRequest({
  url: `https://github-production-user-asset-6210df.s3.amazonaws.com/`,
  method: 'POST',
  data,
  onload(response) {
    console.log(response);
    console.log(response.response);
  },
});
```

400 error message

```xml
<?xml version="1.0" encoding="UTF-8"?>
<Error>
    <Code>
        InvalidArgument
    </Code>
    <Message>
        Bucket POST must contain a field named 'key'.  If it is specified, please check the order of the fields.
    </Message>
    <ArgumentName>
        key
    </ArgumentName>
    <ArgumentValue></ArgumentValue>
    <RequestId>
        SXBZNF1BT1CXSTYS
    </RequestId>
    <HostId>
        Hn+/BwVutY6PaasVat4sOPcM3icjY+kXANNIoybhkH/dnLg4Q2Unha3cYzDHUoSMohpdL7k2y88=
    </HostId>
</Error>
```

**Comment by lisonge** [comment]
@derjanb 

hello, is it not a bug ?

use `GM_xmlhttpRequest` send `FormData` and the order of request body form-data field  is reversed

it should not be reversed, its order should be same as `FormData.prototype.forEach`

**Comment by derjanb** [comment]
This should be fixed at 4.20.6185+ ([crx](https://tampermonkey.net/crx/tampermonkey_beta_current.crx)|xpi in review)

Chrome/Edge users, please download the crx file linked above and drag and drop it to the extensions page `chrome://extensions`.
For a quick fix please export your settings and scripts as zip or (JSON) file at the "Utilities" tab and import it back at the fixed BETA version.
